src/deepsparse/server/monitoring.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import multiprocessing
+import time
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+import pydantic
+import requests
+import yaml
+
+from deepsparse.server.config import EndpointConfig, ServerConfig
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def start_file_watcher(
+    config_path: str, endpoints_url: str, check_interval_s: float
```

Comment:
@bfineran i'm proposing that we monitor the yaml config path instead of a "model directory". I think allowing a model directory in addition to yaml config is confusing from user standpoint. A side benefit is that it's easier to support code wise.